doc/user-guide.adoc

```diff
@@ -312,7 +315,38 @@ spec:
         name: self-signed
     termination: reencrypt
 ----
+==== Using automatic registration with OIDC providers
 
+The operator can request a client Id and client Secret from providers, rather than requiring them in advance. This can simplify deployment, as the provider's administrator can supply the information needed for registration once, instead of clientIds and secrets repetitively.  An additional secret specified by `sso.oidc[].autoRegisterSecret` contains the information necessary to perform the registration.  Automatic registration takes precedence over the values in the sso secret. This is tested with RedHat Single Sign-on (RHSSO). 
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:  
+  name: my-autoreg-secret
+  # Secret must be created in the same namespace as the OpenLibertyApplication instance
+  namespace: demo
+type: Opaque
+data:
+  # base64 encode the data before entering it here.
+  # RHSSO requires an initial access token for registration
+  initialAccessToken: xxxxxyyyyy
+  # others may require a dedicated clientId and clientSecret for registration
+  clientId: bW9vb29vb28=
+  clientSecret: dGhlbGF1Z2hpbmdjb3c=
+  #
+  # Optional: Grant types are the types of OAuth flows the resulting clients will allow
+  # Default is authorization_code,refresh_token.  Specify a comma separated list.
+  # grantTypes: base64 data goes here
+  #
+  # Optional: Scopes limit the types of information about the user that the provider will return.
+  # Default is openid,profile.  Specify a comma-separated list.
+  # scopes: base64 data goes here
+  #
+  # Optional: To skip TLS certificate checking with the provider, specify insecureTLS as true.  Default is false.
+  # insecureTLS: dHJ1ZQ==
+----
```

Comment:
can we also add what the corresponding `RuntimeComponent` CR would be, and how it references this Secret?  It would be good to visualize a working sample of what fields are needed in this auto-registration flow.   Is there a specific naming convention that one must follow for this Secret to be auto discovered, or it can be named whatever we want as long as we reference it from the CR?

Reply:
The secret is an attribute of the OpenLibertyApplication, `sso.oidc[].autoRegisterSecret`, so it's not usable from the RuntimeComponent CR. It can be named whatever we want. I've updated the user guide to make the flow and who needs what (RedHat SSO vs. IBM Cloud Identity) more clear.

Reply:
my bad - I meant to say the corresponding `OpenLibertyApplication` CR sample, that shows a complete CR using `autoRegisterSecret`.     I really like the `Secret` example, I think we just need the corresponding olapp CR.

Reply:
ah, got it.  Added an example